table/block_based/block_based_table_reader.cc

```diff
@@ -981,7 +981,7 @@ void BlockBasedTable::GenerateCachePrefix(Cache* cc, RandomAccessFile* file,
 
   // If the prefix wasn't generated or was too long,
   // create one from the cache.
-  if (cc && *size == 0) {
+  if (*size == 0) {
```

Comment:
Line 966 calls this function with `cc == nullptr`; however, persistent cache does not handle empty cache key prefix, so we should still generate a random one in case `GetUniqueId()` failed. (This change addresses a unit test failure.)

Reply:
tmpfs has all file ID 0. We can handle persistent cache test in the same way on Windows. Or just disable them.

Reply:
Let me check. The most recent appveyor failure did not appear related to this, btw.